odbc/src/api/data.rs

```diff
@@ -443,12 +643,59 @@ mod i32_len {
     ///
     pub unsafe fn set_output_string(
         message: &str,
+        col_num: USmallInt,
+        var_data_indicies: &mut HashMap<USmallInt, (usize, usize)>,
         output_ptr: *mut Char,
         buffer_len: usize,
         text_length_ptr: *mut Len,
     ) -> SqlReturn {
-        let (len, ret) = set_output_string_helper(message, output_ptr, buffer_len);
+        let index = if let Some((index, _)) = var_data_indicies.get(&col_num) {
+            *index
+        } else {
+            0usize
+        };
+        if index >= message.len() {
+            *text_length_ptr = 0;
+            return SqlReturn::NO_DATA;
+        }
+        let (len, ret) =
+            set_output_string_helper(message.get(index..).unwrap(), output_ptr, buffer_len);
+        // the returned length should always be the total length of the data.
+        *text_length_ptr = message.chars().count() as Len;
+        var_data_indicies.insert(col_num, (len + index, 0usize));
```

Comment:
I have a feeling that this is not handling streaming data like it does for set_output_wstring because the total_length is set 0 here and that would make a subsequent call to SQLGetData fail. Or did I miss something?

Reply:
Subsequent calls to `SQLGetData` return SQL_NO_DATA on fixed data. I'm multipurposing the map.

Reply:
Oh right, I don't need the length because we can always get it from `message.len`, which we cannot do for utf-16. I should add a comment to make that clear.